proxy/envoy/ingress.go

```diff
@@ -161,17 +159,19 @@ func generateIngress(conf *IngressConfig) *Config {
 		},
 	}
 
-	// configure for HTTPS if provided with a secret name
-	if conf.Secret != "" {
-		// configure Envoy
+	// configure for HTTPS if there is a secret
+	if secret, err := conf.Secrets.GetSecret("*"); err != nil {
+		glog.Warningf("Error retrieving ingress secret: %v", err)
+	} else if secret != nil {
+		// configure Envoy for HTTPS
 		listener.Address = fmt.Sprintf("tcp://%s:443", WildcardAddress)
```

Comment:
The current approach uses a single listener, either on port 80 **OR** port 443. In practice, there could be several ingress resources where one specifies TLS config, whereas the other doesn't (even for the same vhost). We should be able to distinguish between the two cases, and assign each derived envoy route to either port 80 or 443.

Currently there's no structured way to capture this in the model, but maybe we can add a `RouteRule.Match.Protocol` field for this.